src/pip/_internal/index.py

```diff
@@ -155,7 +166,7 @@ def _get_html_response(url, session):
     if _is_url_like_archive(url):
         _ensure_html_response(url, session=session)
 
-    logger.debug('Getting page %s', url)
+    logger.debug('Getting page %s', _redacted_url(url))
```

Comment:
You should use the `redact_password_from_url()` which is for this purpose.

Reply:
Good to know!, Fixed in the next commit